ksql-engine/src/main/java/io/confluent/ksql/function/udaf/topk/DoubleTopkKudaf.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.function.udaf.topk;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.streams.kstream.Merger;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import io.confluent.ksql.function.KsqlAggregateFunction;
+import io.confluent.ksql.parser.tree.Expression;
+import io.confluent.ksql.util.ArrayUtil;
+import io.confluent.ksql.util.KsqlException;
+
+
+public class  DoubleTopkKudaf extends KsqlAggregateFunction<Double, Double[]> {
+
+  Integer tkVal;
+  Double[] topkArray;
+  Double[] tempTopkArray;
+
+  public DoubleTopkKudaf(Integer argIndexInValue, Integer tkVal) {
+    super(argIndexInValue, new Double[tkVal], SchemaBuilder.array(Schema.FLOAT64_SCHEMA).build(),
+          Arrays.asList(Schema.FLOAT64_SCHEMA),
+          "TOPK", DoubleTopkKudaf.class);
+    this.tkVal = tkVal;
+    this.topkArray = new Double[tkVal];
+    this.tempTopkArray = new Double[tkVal + 1];
+  }
+
+  @Override
+  public Double[] aggregate(Double currentVal, Double[] currentAggVal) {
```

Comment:
Tests for the values this returns under different conditions? There is one test but it only asserts after multiple values have been added to the array. What about when the first value is added?

Reply:
Added tests for this and the other ones.